### PR TITLE
use java.time.Duration in Typed javadsl, #24646

### PR DESCRIPTION
**Pull request by patriknw**
Refs #24646

### CODE REVIEW

**Comment by patriknw** [comment]
tests passed, I didn't run them (thanks Mr. Compiler) :)

**Review by johanandren — APPROVED** [review]
LGTM!

File: akka-actor-typed/src/main/scala/akka/actor/typed/SupervisorStrategy.scala [hunk 1]
```diff
@@ -33,7 +37,7 @@ object SupervisorStrategy {
   val stop: SupervisorStrategy = Stop(loggingEnabled = true)
 
   /**
-   * Restart with a limit of number of restart retries.
+   * Scala API: Restart with a limit of number of restart retries.
```
Comment (chbatey) [inline akka-actor-typed/src/main/scala/akka/actor/typed/SupervisorStrategy.scala]: I wonder if these should be in separate classes/packages for scaladsl/javadsl

**Review by chbatey — APPROVED** [review]
LGTM

**Pull request merged**